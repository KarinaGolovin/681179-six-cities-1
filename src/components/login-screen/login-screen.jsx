import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {SignIn} from '../sign-in/sign-in.jsx';
import {signIn} from '../../store/actions';
import PageTemplate from '../page-template/page-template.jsx';
import withAuthRedirect from '../../hocs/with-auth-redirect/with-auth-redirect';

const SignInWithRedirect = withAuthRedirect(SignIn, {
  authorized: `/`
});

export function LoginScreen(props) {
  const {onSingIn} = props;
  return (
    <PageTemplate>
      <SignInWithRedirect onLogin={onSingIn}/>
    </PageTemplate>
  );
}

const mapDispatchToProps = {
  onSingIn: signIn
};

LoginScreen.propTypes = {
  onSingIn: PropTypes.func,
};

export default connect(
    null,
    mapDispatchToProps
)(LoginScreen);

