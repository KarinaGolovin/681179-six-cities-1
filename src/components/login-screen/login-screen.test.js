import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {LoginScreen} from './login-screen.jsx';

const renderer = new ShallowRenderer();

it(`LoginScreen correctly renders`, () => {
  const component = renderer.render(
      <LoginScreen
        onSingIn={() => {}}/>
  );

  expect(component).toMatchSnapshot();
});
