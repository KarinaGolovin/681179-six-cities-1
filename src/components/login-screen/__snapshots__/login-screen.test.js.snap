// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen correctly renders 1`] = `
<Memo(ConnectFunction)>
  <Memo(ConnectFunction)
    onLogin={[Function]}
  />
</Memo(ConnectFunction)>
`;
